ethers-solc/src/artifacts/ast.rs

```diff
@@ -18,24 +18,47 @@ pub struct Ast {
     pub src: SourceLocation,
     #[serde(default, skip_serializing_if = "Vec::is_empty")]
     pub nodes: Vec<Node>,
+
+    /// Node attributes that were not deserialized.
     #[serde(flatten)]
     pub other: BTreeMap<String, serde_json::Value>,
 }
 
 #[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
 pub struct Node {
+    /// The node ID.
     pub id: usize,
+
+    /// The node type.
     #[serde(rename = "nodeType")]
     pub node_type: NodeType,
+
+    /// The location of the node in the source file.
     #[serde(with = "serde_helpers::display_from_str")]
     pub src: SourceLocation,
+
+    /// Child nodes for some node types.
     #[serde(default, skip_serializing_if = "Vec::is_empty")]
     pub nodes: Vec<Node>,
+
+    /// Body node for some node types.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub body: Option<Box<Node>>,
+
+    /// Node attributes that were not deserialized.
     #[serde(flatten)]
     pub other: BTreeMap<String, serde_json::Value>,
 }
 
-/// Represents the source location of a node : `<start>:<length>:<index>`
+impl Node {
+    /// Deserialize a serialized node attribute.
+    pub fn attribute<D: DeserializeOwned>(&self, key: impl AsRef<str>) -> Option<D> {
+        // TODO: Can we avoid this clone?
+        self.other.get(key.as_ref()).and_then(|v| serde_json::from_value(v.clone()).ok())
```

Comment:
This can probably improved with Serde's tagged enum by doing `enum NodeType { FunctionCall { foo: u8 } }` or whatever, but nbd now